website/public/js/controllers/notificationCtrl.js

```diff
@@ -148,8 +148,9 @@ habitrpg.controller('NotificationCtrl',
     });
 
     // Quest invitation modal
-    $rootScope.$watch('party.quest.key && !party.quest.active && party.quest.members[user._id] == undefined', function(after, before){
-      if (after == before || after != true) return;
+    $rootScope.$watch('user.party.quest.RSVPNeeded == true', function(after, before){
```

Comment:
Adding `party.quest.key` brings us back to the problem this is trying to fix, and the modal won't show up until the page is reloaded (or `fetch recent messages` is pressed on the party page).